core/src/fields.rs

```diff
@@ -469,50 +469,150 @@ impl MongoFields {
     // The query timeout comes from the statement attribute SQL_ATTR_QUERY_TIMEOUT. If there is a
     // timeout, the query must finish before the timeout or an error is returned.
     pub fn list_columns(
-        _mongo_connection: &MongoConnection,
+        mongo_connection: &MongoConnection,
         _query_timeout: Option<i32>,
-        db_name_filter: &str,
-        collection_name_filter: &str,
+        db_name: Option<&str>,
+        collection_name_filter: Option<&str>,
+        field_name_filter: Option<&str>,
     ) -> Self {
+        let dbs = db_name.map_or_else(
+            || {
+                mongo_connection
+                    .client
+                    .list_database_names(None, None)
+                    .unwrap()
+            },
+            |db| vec![db.to_string()],
+        );
         MongoFields {
-            dbs: None,
-            current_db: None,
+            dbs,
+            current_db_name: "".to_string(),
             collections_for_db: None,
-            current_collection: None,
-            num_fields_for_collection: None,
-            current_schema: None,
-            current_field_for_collection: None,
-            db_name_filter: Some(to_name_regex(db_name_filter)),
-            collection_name_filter: Some(to_name_regex(collection_name_filter)),
+            current_col_metadata: Vec::new(),
+            current_field_for_collection: -1,
+            collection_name_filter: collection_name_filter.map(to_name_regex),
+            field_name_filter: field_name_filter.map(to_name_regex),
```

Comment:
I'm looking through the changes and I don't see where the `field_name_filter` is being used to filter the column names.  Is it done elsewhere or maybe it was missed?

Reply:
Good catch, the problem of leaving tests for another patch